packages/ember-metal/lib/meta.js

```diff
@@ -386,60 +357,123 @@ export class Meta {
+    if (ret === undefined) {
+      if (this.parent) {
+        ret = this._chains = this.parent.writableChains(create).copy(this.source);
+      } else {
+        ret = this._chains = create(this.source);
+      }
+    }
+    return ret;
+  }
+
+  readableChains() {
+    return this._getInherited('_chains');
+  }
+
+  writeWatching(subkey, value) {
+   assert(`Cannot call writeWatching after the object is destroyed.`, !this.isMetaDestroyed());
+
+   let map = this._getOrCreateOwnMap('_watching');
+   map[subkey] = value;
+  }
+
+  peekWatching(subkey) {
+   return this._findInherited('_watching', subkey);
+  }
+
+  forEachWatching(fn) {
+    let pointer = this;
+    let seen;
+    while (pointer !== undefined) {
+      let map = pointer._watching;
+      if (map !== undefined) {
+       for (let key in map) {
+         seen = seen || Object.create(null);
+         if (seen[key] === undefined) {
+           seen[key] = true;
+           fn(key, map[key]);
+         }
        }
-     }
-     return ret;
-   }
+    }
+    pointer = pointer.parent;
+    }
+  }
 
-   readableChains() {
-     return this._getInherited('_chains');
-   }
+  clearWatching() {
+   assert(`Cannot call clearWatching after the object is destroyed.`, !this.isMetaDestroyed());
 
-}
+   this._watching = undefined;
+  }
 
-if (EMBER_GLIMMER_DETECT_BACKTRACKING_RERENDER || EMBER_GLIMMER_ALLOW_BACKTRACKING_RERENDER) {
-  Meta.prototype.writableLastRendered = function() { return this._getOrCreateOwnMap('_lastRendered'); };
-  Meta.prototype.readableLastRendered = function() { return this._lastRendered; };
-  if (has('ember-debug')) { //https://github.com/emberjs/ember.js/issues/14732
-    Meta.prototype.writableLastRenderedReferenceMap = function() { return this._getOrCreateOwnMap('_lastRenderedReferenceMap'); };
-    Meta.prototype.readableLastRenderedReferenceMap = function() { return this._lastRenderedReferenceMap; };
-    Meta.prototype.writableLastRenderedTemplateMap = function() { return this._getOrCreateOwnMap('_lastRenderedTemplateMap'); };
-    Meta.prototype.readableLastRenderedTemplateMap = function() { return this._lastRenderedTemplateMap; };
+  deleteFromWatching(subkey) {
+   delete this._getOrCreateOwnMap('_watching')[subkey];
   }
-}
 
-for (let name in listenerMethods) {
-  Meta.prototype[name] = listenerMethods[name];
-}
+  hasInWatching(subkey) {
+   return this._findInherited('_watching', subkey) !== undefined;
+  }
+
+  writeMixins(subkey, value) {
+    assert(`Cannot call writeMixins after the object is destroyed.`, !this.isMetaDestroyed());
+    let map = this._getOrCreateOwnMap('_mixins');
+    map[subkey] = value;
+  }
+
+  peekMixins(subkey) {
+    return this._findInherited('_mixins', subkey);
+  }
+
+  forEachMixins(fn) {
+    let pointer = this;
+    let seen;
+    while (pointer !== undefined) {
+      let map = pointer._mixins;
+      if (map !== undefined) {
+        for (let key in map) {
+          seen = seen || Object.create(null);
```

Comment:
Should we use `create` here too (I think we do `const { create } = Object;` above)?